src/sig/rsasig.c

```diff
@@ -9,18 +9,201 @@
+        DER_SEQUENCE, DER_NIST_HASHALGS_LEN+9+digest_size, \
+          DER_SEQUENCE, DER_NIST_HASHALGS_LEN+5, \
+            DER_OBJECT, DER_NIST_HASHALGS_LEN+1, DER_NIST_HASHALGS, alg_id, \
+            DER_NULL, 0, \
+          DER_OCTET_STRING, digest_size \
+    };
+
+#define DEFINE_DER_RSA_PSS_PARAMS(name, obj_base, obj_alg, obj_len, digest_len) \
+    static const unsigned char der_rsa_pss_params_##name[] = { \
+        DER_SEQUENCE, 25 + DER_RSASSA_PSS_LEN + DER_MGF1_LEN + obj_len + obj_len, \
+            DER_OBJECT, DER_RSASSA_PSS_LEN, DER_RSASSA_PSS, \
+            DER_SEQUENCE, 21 + DER_MGF1_LEN + obj_len + obj_len, \
+                0xA0, 4 + obj_len, \
+                    DER_SEQUENCE, 2 + obj_len, \
+                        DER_OBJECT, obj_len, obj_base, obj_alg, \
+                0xA1, 8 + DER_MGF1_LEN + obj_len, \
+                    DER_SEQUENCE, 6 + DER_MGF1_LEN + obj_len, \
+                        DER_OBJECT, DER_MGF1_LEN, DER_MGF1, \
+                        DER_SEQUENCE, 2 + obj_len, \
+                            DER_OBJECT, obj_len, obj_base, obj_alg, \
+                0xA2, 3, \
+                    DER_INTEGER, 1, digest_len, \
+    };
+
+#define DEFINE_DER_SEQ_SHA(bits, rsa_algid, digestinfo_algid) \
+    static const unsigned char der_rsa_sha##bits[] = { \
+        DER_SEQUENCE, DER_RSADSI_PKCS1_LEN+5, \
+            DER_OBJECT, DER_RSADSI_PKCS1_LEN+1, DER_RSADSI_PKCS1, rsa_algid, \
+            DER_NULL, 0, \
+    }; \
+    DEFINE_DER_DIGESTINFO(sha##bits, digestinfo_algid, bits/8) \
+    DEFINE_DER_RSA_PSS_PARAMS(sha##bits, DER_NIST_HASHALGS, digestinfo_algid, \
+        DER_NIST_HASHALGS_LEN+1, SHA##bits##_DIGEST_LENGTH)
+
+#define DEFINE_DER_SEQ_SHA3(bits, rsa_algid, digestinfo_algid) \
+    static const unsigned char der_rsa_sha3_##bits[] = { \
+        DER_SEQUENCE, DER_NIST_SIGALGS_LEN+5, \
+            DER_OBJECT, DER_NIST_SIGALGS_LEN+1, DER_NIST_SIGALGS, rsa_algid, \
+            DER_NULL, 0 \
+    }; \
+    DEFINE_DER_DIGESTINFO(sha3_##bits, digestinfo_algid, bits/8) \
+    DEFINE_DER_RSA_PSS_PARAMS(sha3_##bits, DER_NIST_HASHALGS, digestinfo_algid, \
+        DER_NIST_HASHALGS_LEN+1, SHA##bits##_DIGEST_LENGTH)
+
+/* ... pkcs(1) 10 (id-RSASSA-PSS) */
+#define DER_RSASSA_PSS DER_RSADSI_PKCS1, 0x0A
+#define DER_RSASSA_PSS_LEN (DER_RSADSI_PKCS1_LEN + 1)
+
+/* ... pkcs(1) 8 (id-RSASSA-PSS) */
+#define DER_MGF1 DER_RSADSI_PKCS1, 0x08
+#define DER_MGF1_LEN (DER_RSADSI_PKCS1_LEN + 1)
+
+static const unsigned char der_rsa_sha1[] = {
+    DER_SEQUENCE, DER_RSADSI_PKCS1_LEN+5,
+        DER_OBJECT, DER_RSADSI_PKCS1_LEN+1, DER_RSADSI_PKCS1, 0x05,
+        DER_NULL, 0
+};
+
+/* iso(1) org(3) oiw(14) secsig(3) algorithms(2) */
+#define DER_SECSIG_ALGO 1 * 40 + 3, 14, 3, 2
+#define DER_SECSIG_ALGO_LEN 4
+/* iso(1) org(3) oiw(14) secsig(3) algorithms(2) hashAlgorithmIdentifier(26) */
+#define ID_SHA1 DER_SECSIG_ALGO, 26
+#define ID_SHA1_LEN DER_SECSIG_ALGO_LEN+1
```

Comment:
clang format is off here, but I would still try to keep it readable by adding spaces around operators:
```suggestion
#define ID_SHA1_LEN DER_SECSIG_ALGO_LEN + 1
```